test/integration/deployment/deployment_test.go

```diff
@@ -381,3 +381,73 @@ func TestDeploymentHashCollision(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestOverlappingDeployments(t *testing.T) {
+	s, closeFn, rm, dc, informers, c := dcSetup(t)
+	defer closeFn()
+	name := "test-overlapping-deployments"
+	ns := framework.CreateTestingNamespace(name, s, t)
+	defer framework.DeleteTestingNamespace(ns, s, t)
+
+	replicas := int32(1)
+	firstDeploymentName := "first-deployment"
+	secondDeploymentName := "second-deployment"
+	testers := []*deploymentTester{
+		{t: t, c: c, deployment: newDeployment(firstDeploymentName, ns.Name, replicas)},
+		{t: t, c: c, deployment: newDeployment(secondDeploymentName, ns.Name, replicas)},
+	}
+
+	// Start informer and controllers
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	informers.Start(stopCh)
+	go rm.Run(5, stopCh)
+	go dc.Run(5, stopCh)
+
+	// Create 2 deployments with overlappings selectors
+	var err error
+	for _, tester := range testers {
+		tester.deployment, err = c.ExtensionsV1beta1().Deployments(ns.Name).Create(tester.deployment)
+		dname := tester.deployment.Name
+		if err != nil {
+			t.Fatalf("failed to create deployment %q: %v", dname, err)
+		}
+		// Wait for the deployment to be updated to revision 1
+		if err = tester.waitForDeploymentRevisionAndImage("1", fakeImage); err != nil {
+			t.Fatalf("failed to update deployment %q to revision 1: %v", dname, err)
+		}
+		// Make sure the deployment's status becomes valid while manually marking its pods as ready simultaneously
+		if err = tester.waitForDeploymentStatusValidAndMarkPodsReady(); err != nil {
+			t.Fatalf("deployment %q failed to have its status becomes valid: %v", dname, err)
+		}
+		// Wait for the deployment to complete
+		if err = tester.waitForDeploymentCompletes(); err != nil {
+			t.Fatalf("deployment %q failed to complete: %v", dname, err)
+		}
+	}
+
+	// Both deployments should proceed independently
+	rsSelector, err := metav1.LabelSelectorAsSelector(testers[0].deployment.Spec.Selector)
+	if err != nil {
+		t.Fatalf("failed to parse deployment selector: %v", err)
+	}
+	rsList, err := c.ExtensionsV1beta1().ReplicaSets(ns.Name).List(metav1.ListOptions{LabelSelector: rsSelector.String()})
+	if err != nil {
+		t.Fatalf("failed to list all replicasets in namespace %q", ns.Name)
```

Comment:
print `err` also, and change "all replicasets" to "replicasets of deployments X and Y..."